### ISSUE TITLE
Improve performance of DigraphSymmetricClosure and OutNeighbours on mutable digraphs

### ISSUE THREAD

**Issue by reiniscirpons** [digraphs/Digraphs#413]
Consider the following timing comparison for taking the symmetric closure of an immutable and mutable version of a digraph:
```
gap> DigraphSymmetricClosure(CycleDigraph(IsImmutableDigraph, 10000));; time;
33
gap> DigraphSymmetricClosure(CycleDigraph(IsMutableDigraph, 10000));; time;
6174
```
Creating the mutable and immutable version of the cycle graph takes about the same time, and the cycle digraph is not symmetric, so the time discrepancy is not due to the immutable version returning a stored attribute. This is an example of a mutable digraph taking more time on a simple modification than its immutable counterpart, which goes against the main purpose of mutable digraphs - to provide efficient modification operations.

According to  #318, the issue is most likely due to  `OutNeighbours` being copied every time it is accessed, as it is an attribute not a function, and mutable digraphs do not store attributes. I would argue that, since `OutNeighbours` (or some similar variable storing neighbour information) is dynamically updated as the mutable digraph changes, allowing more efficient access to it is essential, since some nontrivial modifications we may want to make require knowledge of a vertices local neighbourhood (for example, swapping edges, or taking symmetric closure).

This issue comes in relation to the current named graphs project, part of which requires implementing infinite families of graphs - in some cases the graphs can be created by taking products of existing structures and making small modifications to them (for example, swapping some edges in the product etc), which is a prime use case for the mutable graphs, and would benefit from the speedup associated to making `OutNeighbours` more efficient, and the graphs we are implementing also generally tend to be symmetric, so having `DigraphSymmetricClosure` be faster also helps.

**Comment by wilfwilson** [comment]
For me, oon the `stable-1.4` and `master` branches now:
```
gap> DigraphSymmetricClosure(CycleDigraph(IsImmutableDigraph, 10000));; time;
14
gap> DigraphSymmetricClosure(CycleDigraph(IsMutableDigraph, 10000));; time;
17
```
(Roughly). Aka resolved!

**Comment by wilfwilson** [comment]
Resolved in v1.4.1.
